src/main/java/executor/service/service/impl/DIFactory.java

```diff
@@ -32,42 +32,56 @@
+        if (instance != null) {
+            return instance;
+        }
+        synchronized (DIFactory.class) {
+            if (instance == null) {
+                instance = new DIFactory();
+            }
+        }
+        return instance;
+    }
+
+    @Override
+    public <T> T createObject(Class<T> clazz) {
         PropertiesConfig propertiesConfig = new PropertiesConfig();
         ProxyProvider proxyProvider = new JSONFileProxyProvider();
-        BeanConfig beanConfig = new BeanConfig(propertiesConfig);
-        this.threadPoolExecutor = beanConfig.threadPoolExecutor();
-        WebDriverConfig webDriverConfig = beanConfig.webDriverConfig();
+        WebDriverConfig webDriverConfig = new WebDriverConfigObject(propertiesConfig).webDriverConfig();
         StepExecutionClickCss stepExecutionClickCss = new StepExecutionClickCssImpl();
         StepExecutionSleep stepExecutionSleep = new StepExecutionSleepImpl();
         StepExecutionClickXpath stepExecutionClickXpath = new StepExecutionClickXpathImpl();
-        ScenarioExecutor scenarioExecutor = new ScenarioExecutorImpl(
-                stepExecutionClickCss,
-                stepExecutionSleep,
-                stepExecutionClickXpath);
-        this.executionService = new ExecutionServiceImpl(scenarioExecutor, webDriverConfig, new WebDriverInitializerImpl(propertiesConfig));
+        ScenarioExecutor scenarioExecutor = new ScenarioExecutorImpl(stepExecutionClickCss,stepExecutionSleep,stepExecutionClickXpath);
         ScenarioProvider scenarioProvider = new JSONFileScenarioProvider();
-        this.scenarioSourceListener = new ScenarioSourceListenerImpl(scenarioProvider, propertiesConfig);
+        ScenarioSourceListener scenarioSourceListener = new ScenarioSourceListenerImpl(scenarioProvider, propertiesConfig);
         ProxyValidator proxyValidator = new ProxyValidatorImpl(propertiesConfig);
-        this.proxySourcesClient = new ProxySourcesClientImpl(proxyProvider, proxyValidator, propertiesConfig);
-        this.tasksFactory = new TasksFactoryImpl();
-    }
+        ProxySourceClient proxySourceClient = new ProxySourceClientImpl(proxyProvider, proxyValidator, propertiesConfig);
+        ScenarioSourceQueue scenarioSourceQueue = new ScenarioSourceQueue();
+        ScenarioTaskWorker scenarioTaskWorker = new ScenarioTaskWorker(scenarioSourceListener, scenarioSourceQueue);
+        ProxySourceQueue proxySourceQueue = new ProxySourceQueue();
+        ProxyTaskWorker proxyTaskWorker = new ProxyTaskWorker(proxySourceClient, proxySourceQueue);
+        WebDriverInitializer webDriverInitializer = new WebDriverInitializerImpl(propertiesConfig);
+        ExecutorService threadPoolExecutor = new ThreadPoolExecutor(propertiesConfig).threadPoolExecutor();
+        ExecutionService executionService = new ExecutionServiceImpl(scenarioExecutor, webDriverConfig, webDriverInitializer);
+        TasksFactory tasksFactory = new TasksFactoryImpl(scenarioTaskWorker, proxyTaskWorker);
```

Comment:
If you decide to cache them, to make their initialization at the same time of creating object.
Also you didn't decide problem of multithreading.

 ProxySourceClient proxySourceClient =  map.put(new ....);

Reply:
Thanks.
Will be used:
private final Map<Class<?>, Object> cache = new ConcurrentHashMap<>();